src/pages/index.js

```diff
@@ -14,71 +14,154 @@ import Section from "../components/Section.js";
 
 import UserInfo from "../components/UserInfo.js";
 
+import Api from "../components/Api.js";
+
 import {
-  initialCards,
   config,
   editButton,
   addButton,
-  profileNameInput,
-  profileDescriptionInput,
   addModalForm,
-  cardTitleInput,
-  cardURLInput,
+  profileAvatarContainer,
+  profileEditForm,
+
 } from "../utils/constants.js";
 
+// Initialize API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c91d1d7e-9726-43c4-acb8-289510ab9429",
+    "Content-Type": "application/json",
+  },
+});
+
 // Initialize components
 
 const userInfo = new UserInfo({
   profileNameSelector: "#profile__name",
   profileJobSelector: "#profile__description",
+  avatarSelector: "#profile__pfp",
 });
 
 const profilePopup = new PopupWithForm("#edit-modal", (formData) => {
-  userInfo.setUserInfo(formData);
-  profilePopup.close();
+  profilePopup.renderLoading(true);
+  return api
+    .updateUserInfo(formData)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profilePopup.close();
+      formValidators["edit-modal__form"].resetValidation();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.renderLoading(false);
+    });
 });
 profilePopup.setEventListeners();
 
 const addCardPopup = new PopupWithForm("#add-modal", (formData) => {
-  handleAddCardSubmit(formData);
+  addCardPopup.renderLoading(true);
+  return api
+    .createCard(formData)
+    .then((res) => {
+      const { name, link, _id } = res;
+      const cardElement = createCard({ name, link, _id });
+      cardSection.addItem(cardElement);
+      addCardPopup.close();
+      addModalForm.reset();
+      formValidators["add-modal__form"].resetValidation();
```

Comment:
`resetValidation` is for clearing errors but you don't have them in block `then`. Please, call `disableButton` here